app/src/main/java/chat/rocket/android/server/domain/SettingsRepository.kt

```diff
@@ -84,13 +83,16 @@ fun PublicSettings.allowedMessagePinning(): Boolean = this[ALLOW_MESSAGE_PINNING
 fun PublicSettings.allowedMessageEditing(): Boolean = this[ALLOW_MESSAGE_EDITING]?.value == true
 fun PublicSettings.allowedMessageDeleting(): Boolean = this[ALLOW_MESSAGE_DELETING]?.value == true
 
-fun PublicSettings.uploadMimeTypeFilter(): Array<String> {
-    val values = this[UPLOAD_WHITELIST_MIMETYPES]?.value
-    values?.let { it as String }?.split(",")?.let {
-        return it.mapToTypedArray { it.trim() }
+fun PublicSettings.uploadMimeTypeFilter(): Array<String>? {
+    val values = this[UPLOAD_WHITELIST_MIMETYPES]?.value as String?
+    if (!values.isNullOrBlank()){
+        // It's far more efficient to split the string to an array using the Java type rather than the Kotlin type
+        @Suppress("PLATFORM_CLASS_MAPPED_TO_KOTLIN")
+        val filter = (values as java.lang.String).split(",")
```

Comment:
On this case I'd prefer readability over performance. How much difference this will make here? Doing a simple measure, the gain seems negligible.
`return values.split(",").mapToTypedArray { it.trim() }` is more readable and idiomatic.